functions/src/models/requests/index.ts

```diff
@@ -299,6 +300,22 @@ export const RequestFirestoreConverter: FirestoreDataConverter<Request> = {
     return Request.factory(data.data());
   },
   toFirestore: (modelObject: Request): DocumentData => {
-    return modelObject.toObject();
+    return {
+      cavUserRef: modelObject.cavUserRef,
+      cavUserSnapshot: modelObject.cavUserSnapshot ? modelObject.cavUserSnapshot.toObject() : null,
+      pinUserRef: modelObject.pinUserRef,
+      pinUserSnapshot: modelObject.pinUserSnapshot.toObject(),
+      title: modelObject.title,
+      description: modelObject.description,
+      latLng: JSON.stringify(modelObject.latLng),
+      streetAddress: modelObject.streetAddress,
+      status: modelObject.status,
+      createdAt: modelObject.createdAt.toDate(),
+      updatedAt: modelObject.updatedAt.toDate(),
+      pinRating: modelObject.pinRating,
+      cavRating: modelObject.cavRating,
+      pinRatedAt: modelObject.pinRatedAt?.toDate() || null,
+      cavRatedAt: modelObject.cavRatedAt?.toDate() || null,
+    };
```

Comment:
Is there any reason we moved away from `model.toObject` here? We don't want to have to maintain this code in two functions.

Reply:
https://github.com/reach4help/reach4help/pull/611#discussion_r433959646
apart from what I mentioned in the comment linked here, I also have the issue of latLng, to which I was absolutely not able to figure out any solution at all apart from serializing.
But this will destroy the data structure and the frontend would get values that I can't use.
since I am using the `toObject` method in the cloud function, I didn't want to touch it.

This is purely for fixing the errors in the tests